### DISCUSSION TITLE
match statement

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#15256] [General]
How likely is it that the Python match-statememt will ever be included in microPython?

**Comment by peterhinch** [comment]
PEP636 has been discussed - the consensus seemed to be "unlikely". I gather it requires a different parser and the implementation is unlikely to be "micro".

**Comment by mmliam** [comment]
I suspected something like that; pity. thanks.

**Comment by sosi-deadeye** [comment]
A bit of history.

The new PEG parser [(PEP 617)](https://peps.python.org/pep-0617/) were released with CPython 3.9. This change allowed to implement the walrus operator (assignment expression).

The structural pattern matching [(PEP 622)](https://peps.python.org/pep-0622/) was introduced with CPython 3.10 and was the first use of a soft keyword.

Quote from the PEP:

> The match statement uses a soft keyword, and it is one of the first major Python features to take advantage of the capabilities of the new PEG parser. This means that third-party parsers which are not ‘PEG-compatible’ will have a hard time with the new syntax.

My implication is, if a `LL(1)` parser is used, the `match` statement must become a keyword. Then you could not use `match` as a name. Very bad.

If someone could write a PEG parser for micropython, which is in size equal or smaller, this would be a nice addition.
